R/two.step.hybrid.R

```diff
@@ -1,170 +1,412 @@
-			
-			r$this.ftrs
-		}
-		
-		colnames(new.this.aligned)<-colnames(this.aligned)
-		
-		####
-		
-		if(batch.i == 1)
-		{
-			aligned<-new.this.aligned
-		}else{
-			aligned<-cbind(aligned, new.this.aligned)
-		}
+        this.aligned[sample, ] <- recaptured
+        this.pk.time[sample, ] <- recaptured.time
+      }
     }
-    
-    aligned<-cbind(fake3$aligned.ftrs[,1:4], aligned)
-    
-	batch.presence.mat<-matrix(0, nrow=nrow(aligned), ncol=length(batches))
-	for(batch.i in 1:length(batches))
-	{
-		batch<-batches[batch.i]
-        this.mat<-aligned[,which(colnames(aligned) %in% info[info[,2]==batch,1])]
-		this.mat<- 1*(this.mat != 0)
-		this.presence<-apply(this.mat,1,sum)/ncol(this.mat)
-		batch.presence.mat[,batch.i]<-1*(this.presence >= min.within.batch.prop.report)
-	}
-	batch.presence<-apply(batch.presence.mat,1,sum)/ncol(batch.presence.mat)
-	final.aligned<-aligned[which(batch.presence >= min.batch.prop),]
-	
-    stopCluster(cl)
-    to.return<-new("list")
-    to.return$batchwise.results<-batchwise
-	to.return$all.detected.ftrs<-aligned
-    to.return$final.ftrs<-final.aligned
-    return(to.return)
+
+    colnames(this.aligned) <- extract_pattern_colnames(this.fake, "_intensity")
+    colnames(this.aligned) <- stringr::str_remove_all(colnames(this.aligned), "_intensity")
+    colnames(this.pk.time) <- extract_pattern_colnames(this.fake, "_rt")
+    colnames(this.pk.time) <- stringr::str_remove_all(colnames(this.pk.time), "_rt")
+
+
+    aligned_features <- dplyr::select(aligned, mz, rt, mz_min, mz_max)
+    aligned_int_crosstab <- dplyr::bind_cols(aligned_features, this.aligned)
+    aligned_rt_crosstab <- dplyr::bind_cols(aligned_features, this.pk.time)
+    recovered_batchwise <- recover_weaker_signals(
+      cluster = cluster,
+      filenames = filter(filenames_batchwise, batch == batch_id)$filename,
+      extracted_features = batchwise[[batch_id]]$extracted_features,
+      corrected_features = batchwise[[batch_id]]$corrected_features,
+      aligned_rt_crosstab = aligned_rt_crosstab,
+      aligned_int_crosstab = aligned_int_crosstab,
+      original_mz_tolerance = mz.tol,
+      aligned_mz_tolerance = batch.align.mz.tol,
+      aligned_rt_tolerance = batch.align.chr.tol,
+      mz_range = recover.mz.range,
+      rt_range = recover.chr.range,
+      use_observed_range = use.observed.range,
+      min_bandwidth = min.bw,
+      max_bandwidth = max.bw,
+      recover_min_count = recover.min.count
+    )
+
+    recovered_batchwise <- as_wide_aligned_table(recovered_batchwise)
+
+    recovered <- dplyr::full_join(recovered, recovered_batchwise, by = c("mz", "rt", "mz_min", "mz_max"))
+  }
+  recovered <- dplyr::select(recovered, mz, rt, mz_min, mz_max, contains("_rt"), contains("_intensity"))
+
+  return(recovered)
+}
+
+batchwise_wrapper <- function(batchwise, batches_idx) {
```

Comment:
The function name could be a bit more comprehensive - `batchwise` what? what is wrapped here?